core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/DatasourceService.java

```diff
@@ -154,4 +171,54 @@ public static boolean isDSWDatasource( Domain domain ) {
     }
     return !isMetadataDatasource( domain );
   }
+
+  public boolean isDSWDatasource( String domainId ) {
+    return !isMetadataDatasource( domainId );
+  }
+
+  protected int getDatasourceLoadThreadCount() throws DatasourceServiceException {
+    int threadCount = Runtime.getRuntime().availableProcessors();
+    String threadCountAsString =
+        pluginResourceLoader.getPluginSetting( getClass(), "settings/data-access-datasource-load-threads" );
+    if ( StringUtils.isNotBlank( threadCountAsString ) ) {
+      threadCount = Integer.parseInt( threadCountAsString );
```

Comment:
What about `NumberFormatException`?

Reply:
I would also explicitly throw `NumberFormatException` for the `<=0` test. As far as I know, this exception is also used for "out of range" cases.

Reply:
So my new change will have a single catch block of NumberFormationException in getDatasourceIds() method to handle both the cases for <=0 and String values.